app/assets/v2/js/pages/bounty_details.js

```diff
@@ -204,30 +206,103 @@ var pendingChangesWarning = function(issueURL, last_modified_time_remote, now){
                 }
              });
         };
-        var check_for_bounty_changed_updates_web3 = function(){
-            callFunctionWhenTransactionMined(localStorage['txid'],function(){
-                var bounty = web3.eth.contract(bounty_abi).at(bounty_address());
-                setTimeout(function(){
-                    bounty.bountydetails.call(issueURL, function(error, result){
-                        if(error){
-                            setTimeout(check_for_bounty_changed_updates_web3, 1000);
+
+        // Only run this after the transaction is confirmed to be on the blockchain
+        var getBountyId = function (callback) {
+            // Get total number of bounties on the contract
+            if (localStorage['bountyId']) {
+                callback(null, localStorage['bountyId']);
+            }
+            var transactionInfo;
+            var bountiesLength;
+            var bountyId;
+            // var bounty = web3.eth.contract(bounty_abi).at(bounty_address());
+            var bounty = web3.eth.contract(bounty_abi).at(bounty_address());  //hardcode to test
+
+            bounty.getNumBounties(function(error, result){
```

Comment:
a lot of code to just get the bountyID :)

i wish there was an easier way.

Reply:
Yea I know :( . I am not a fan of it.